oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java

```diff
@@ -201,22 +241,123 @@ public APIGatewayProxyResponseEvent authoriseRequestHandler(
                     persistentSessionId,
                     authRequest.getClientID().getValue(),
                     clientSessionId);
-        } else {
-            authRequest = RequestObjectToAuthRequestHelper.transform(authRequest);
-            return getOrCreateSessionAndRedirect(
-                    sessionService.getSessionFromSessionCookie(input.getHeaders()),
+        }
+        Optional<Session> session = sessionService.getSessionFromSessionCookie(input.getHeaders());
+        if (configurationService.isDocAppDecoupleEnabled()
+                && DocAppUserHelper.isDocCheckingAppUser(
+                        authRequest.toParameters(), Optional.of(client))) {
+            return handleDocAppJourney(
+                    session,
                     authRequest,
+                    client,
+                    clientSessionId,
                     ipAddress,
-                    persistentSessionId,
-                    clientSessionId);
+                    PersistentIdHelper.extractPersistentIdFromHeaders(input.getHeaders()));
         }
+
+        authRequest = RequestObjectToAuthRequestHelper.transform(authRequest);
+        return handleAuthJourney(
+                session, authRequest, ipAddress, persistentSessionId, client, clientSessionId);
+    }
+
+    private APIGatewayProxyResponseEvent handleDocAppJourney(
+            Optional<Session> existingSession,
+            AuthenticationRequest authRequest,
+            ClientRegistry client,
+            String clientSessionId,
+            String ipAddress,
+            String persistentId) {
+
+        var session = existingSession.orElseGet(sessionService::createSession);
+        attachSessionIdToLogs(session);
+
+        if (existingSession.isEmpty()) {
+            updateAttachedSessionIdToLogs(session.getSessionId());
+            LOG.info("Created session");
+        } else {
+            var oldSessionId = session.getSessionId();
+            sessionService.updateSessionId(session);
+            updateAttachedSessionIdToLogs(session.getSessionId());
+            LOG.info("Updated session id from {} - new", oldSessionId);
+        }
+
+        var clientName = client.getClientName();
+
+        Subject subjectId =
+                DocAppSubjectIdHelper.calculateDocAppSubjectId(
+                        authRequest.toParameters(),
+                        configurationService.isCustomDocAppClaimEnabled(),
+                        configurationService.getDocAppDomain());
+        LOG.info("Doc app request received");
+
+        var clientSession =
+                clientSessionService.generateClientSession(
+                        authRequest.toParameters(),
+                        LocalDateTime.now(),
+                        orchestrationAuthorizationService.getEffectiveVectorOfTrust(authRequest),
+                        clientName);
```

Comment:
Similar to the above comment, we do this in both the `handleDocAppJourney` and `handleAuthJourney` methods so wonder if we could move this before these methods are invoked to avoid duplication